prov/gni/src/gnix_vc.c

```diff
@@ -599,6 +600,7 @@ static int __gnix_vc_hndl_conn_resp(struct gnix_cm_nic *cm_nic,
 	 */
 
 	vc->conn_state = GNIX_VC_CONNECTED;
+	vc->peer_id = peer_id;
```

Comment:
There's another path where it would be worth setting the peer_id.  I'll add that to this PR.